### PR TITLE
Wrap all routes in toStrictEntity

### LINKED ISSUE #1030: Inconsistent responses to /connect API
### Description
Responses to the API '/connect' are not reliable, most time it returns :
`
400 BAD REQUEST
{
    "error": "The form field 'uri' was malformed:\nSubstream Source cannot be materialized more than once"
}`

All the performed request are equal.

### Expected behaviour

* 200 OK "connected" at the first time it is called
* 200 OK "already connected" all the following times

### Actual behaviour

The first call of the API once the node is started is consistent: it returns "connected".
The following calls response are inconsistent, some undetected internal behaviour makes them vary between:
* 200 OK "already connected" all the following times
* 400 BAD REQUEST `{    "error": "The form field 'uri' was malformed:\nSubstream Source cannot be materialized more than once"}`

### How to reproduce

The issue can be reproduced simulating multiple calls to the connect API. I attach the postman and the CURL version I used during my tests. 
(The basic token matches )

curl -u :changeit -X POST -F uri=02acb5b3bdf46d3a35e665c9abe46def12c341f55d603853ddeb1371076cab5c57@68.183.221.146:49735 localhost:8080/connect

(My apologies to the guy running this node)

### Eclair Version and environment 
Eclair 2.11 Self-compiled from Github Master branch using: 
* Scala 2.12.8
* Java 11, openjdk 11.0.2 2019-01-15
* MacOs 10.14.5

### Extra information:

It looks like that if the call is not performed for more than 1 minutes, the following call will return the correct message "already connected".

Eclair.log does not contain any line regarding this connection and nothing is printed once an API call is performed.

**eclair.conf**:

`eclair.bitcoind.rpcuser="foo"
eclair.bitcoind.rpcpassword="bar"
eclair.bitcoind.rpcport="34897"
eclair.bitcoind.host="127.0.0.1"
eclair.chain="testnet"
eclair.api.enabled=true
eclair.api.password="changeit"`

### PR DESCRIPTION
**Pull request by araspitzu**
This PR fixes #1030 by ensuring that the API parameters are parsed in a _strict_ manner so without partially reading the data and stopping but ensuring that once we start reading we get the full package even if then it is rejected by the unmarshaller. The issue arises mostly when there are concurrent calls to the same endpoint and is a known behavior of akka, see [issue](https://github.com/akka/akka-http/issues/745).

### CODE REVIEW

**Comment by sstone** [comment]
I think it's the right solution, just taking a bit more time to review it.

**Comment by sstone** [comment]
Using `toStrictEntity` seems to be the "standard" solution to this problem (see comments on https://groups.google.com/forum/#!topic/akka-user/v-jNYPkKVCU, https://github.com/akka/akka-http/issues/117) but there seems to be a performance impact (see https://github.com/eclipse/ditto/issues/412) and some of the suggested fixes which are nearly 2 years old have not been merged. 

I think using `toStrictEntity`, but with a shorter time-out (5 seconds instead of 30, since all our API requests are small) should be fine, but another option is just to avoid matching on multiple formField extractors and do something like:

``` scala
                      path("connect") {
                        formFields("uri".as[NodeURI].?, nodeIdFormParam.?, "host".as[String].?, "port".as[Int].?) { (uri_opt, nodeId_opt, host_opt, port_opt) =>
                          (uri_opt, nodeId_opt, host_opt, port_opt) match {
                            case (Some(uri), None, None, None) => complete(eclairApi.connect(Left(uri)))
                            case (None, Some(nodeId), Some(host), _) => complete(eclairApi.connect(Left(NodeURI(nodeId, HostAndPort.fromParts(host, port_opt.getOrElse(NodeURI.DEFAULT_PORT))))))
                            case (None, Some(nodeId), None, None) => complete(eclairApi.connect(Right(nodeId)))
                            case _ => reject(MalformedFormFieldRejection("connect", "You must either provide field 'uri' or field 'nodeId' with optional fields 'host' and 'port'"))
                          }
                        }
```

This is much closer to what we have and there's only 1 or 2 places where we would need to do this. WDYT ?

**Comment by araspitzu** [comment]
I like the suggestion of not matching on multiple `formParams` if we can do that it's better than the workaround.

**Comment by pm47** [comment]
> another option is just to avoid matching on multiple formField extractors and do something like:

It's less elegant than what we have now, but if that solves the issue and saves us from using timeouts, then I'm all for it.

**Comment by araspitzu** [comment]
[`withChannelIdentifier`](https://github.com/ACINQ/eclair/blob/master/eclair-core/src/main/scala/fr/acinq/eclair/api/ExtraDirectives.scala#L52) makes it quite hard to apply the change, especially in places like [/close](https://github.com/ACINQ/eclair/blob/master/eclair-core/src/main/scala/fr/acinq/eclair/api/Service.scala#L167) we do match multiple `formFields` because of the channel identifier extraction that has been factored out

**Comment by sstone** [comment]
And I missed that there is another form field for the API timeout....

File: eclair-core/src/main/scala/fr/acinq/eclair/api/Service.scala [hunk 1]
```diff
@@ -70,6 +70,8 @@ trait Service extends ExtraDirectives with Logging {
   implicit val actorSystem: ActorSystem
   implicit val mat: ActorMaterializer
 
+  val paramParsingTimeout = 30 seconds
```
[the remark this answers is not in the document]
Comment (araspitzu) [reply eclair-core/src/main/scala/fr/acinq/eclair/api/Service.scala]: To be specific this timeout puts a limit on the time spent receiving data of a request parameter, ACK your suggestion I'll put a smaller timeout and a comment.

**Comment by pm47** [comment]
Looks ok, hope there won't be side effects !

**Pull request merged**